src/gentropy/datasource/finngen_ukb_meta/study_index.py

```diff
@@ -0,0 +1,62 @@
+"""Study Index for Finngen data source."""
+from __future__ import annotations
+
+import pyspark.sql.functions as f
+from pyspark.sql import SparkSession
+
+from gentropy.dataset.study_index import StudyIndex
+
+
+class FinngenUkbMetaStudyIndex(StudyIndex):
+    """Study index dataset from FinnGen UKB meta-analysis."""
+
+    @classmethod
+    def from_source(
+        cls: type[FinngenUkbMetaStudyIndex],
+        spark: SparkSession,
+        raw_study_index_path: str,
+    ) -> StudyIndex:
+        """This function ingests study level metadata from FinnGen UKB meta-analysis.
+
+        Args:
+            spark (SparkSession): Spark session object.
+            raw_study_index_path (str): Raw study index path.
+
+        Returns:
+            StudyIndex: Parsed and annotated FinnGen UKB meta-analysis study table.
+        """
+        # Read the raw study index and process.
+        study_index_df = (
+            spark.read.csv(raw_study_index_path, sep="\t", header=True)
+            .select(
+                f.lit("gwas").alias("studyType"),
+                f.lit("FINNGEN_R11_UKB_META").alias("projectId"),
```

Comment:
Is this datasource going to be versionned with "finngen" releases?

Reply:
Yes — so in other words, if R12 comes out, these studies should have different names (R12 instead of R11). At least that's my understanding currently.

I've added this to the list of questions I have to be discussed at the Core meeting tomorrow